src/Agent.Worker/StepsRunner.cs

```diff
@@ -67,6 +67,13 @@ public sealed class StepsRunner : AgentService, IStepsRunner
                 {
                     await step.RunAsync();
                 }
+                catch (OperationCanceledException ex)
+                {
+                    // Log the error and fail the step.
```

Comment:
will update the comment. I'll do other changes to set job level result code to cancel and break the loop to prevent next steps from running.